### PR TITLE
Make `cargo build` work

### PR DESCRIPTION
**Pull request by matthiasgoergens**
Right now a simple `cargo build` fails.  So let's fix that.

This PR also disentangles `ceno_rt`, so that we can work on building guest programs much easier.

Update: I extracted the remove of the target matrix into https://github.com/scroll-tech/ceno/pull/725 for ease of review.

This PR also enables debug assertions for our tests.

### CODE REVIEW

**Comment by matthiasgoergens** [comment]
Tests are failing, because they are broken on master.

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -32,13 +32,6 @@ jobs:
     timeout-minutes: 30
     runs-on: [self-hosted, Linux, X64]
 
-    strategy:
```
Comment (matthiasgoergens) [inline .github/workflows/integration.yml:35]: Our strategy matrix only has one entry.  So we might as well not bother.

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -50,16 +43,17 @@ jobs:
             ~/.cargo/registry/cache/
             ~/.cargo/git/db/
             target/
+            guest/target/
           key: integration-${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
       - uses: dtolnay/rust-toolchain@nightly
 
       - name: Run example
         env:
-          RAYON_NUM_THREADS: 2
```
Comment (matthiasgoergens) [inline .github/workflows/integration.yml:58]: This isn't read anywhere.

File: .github/workflows/integration.yml (hunk 1 above)
Comment (matthiasgoergens) [reply .github/workflows/integration.yml:35]: Extracted into https://github.com/scroll-tech/ceno/pull/725

File: guest/examples/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[package]
+categories.workspace = true
+description = "Ceno runtime library"
```
Comment (lispc) [inline guest/examples/Cargo.toml:3]: Ceno runtime library => Ceno examples

File: Makefile.toml [hunk 4]
```diff
@@ -11,10 +9,7 @@ args = [
   "--bins",
   "--tests",
   "--examples",
-  "--release",
```
Comment (lispc) [inline Makefile.toml:14]: not sure whether it is good to remove the `--release`.   Feel free to close this if that is ok.

**Review by lispc — APPROVED** [review]
only some small comments

File: Makefile.toml (hunk 4 above)
Comment (matthiasgoergens) [reply Makefile.toml:14]: Good spotting.  In this case, it's what we want to do:

`--release` disables debug assertions.  Recently we had a problem with [debug assertions being broken](https://github.com/scroll-tech/ceno/issues/715).  And we already [enabled them for integration tests](https://github.com/scroll-tech/ceno/pull/714/files).

File: guest/examples/Cargo.toml (hunk 3 above)
Comment (matthiasgoergens) [reply guest/examples/Cargo.toml]: Good catch!

File: Makefile.toml (hunk 4 above)
Comment (matthiasgoergens) [reply Makefile.toml:14]: In compensation, we upped the default optimisation level for the dev-build to 2.  As a compromise between speed of building vs speed of execution.

File: examples/.cargo/config.toml
Comment (matthiasgoergens) [inline examples/.cargo/config.toml:1]: This file moved, it wasn't deleted.  Command line git shows that, but not github.  Alas.

**Pull request merged**